packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -202,40 +378,112 @@ class Actions extends InheritedWidget {
   /// {@endtemplate}
-  final Map<LocalKey, ActionFactory> actions;
+  final Map<Type, Action<Intent>> actions;
+
+  /// {@macro flutter.widgets.child}
+  final Widget child;
+
+  // Visits the Actions widget ancestors of the given element using
+  // getElementForInheritedWidgetOfExactType. Returns true if the visitor found
+  // what it was looking for.
+  static bool _visitActionsAncestors(BuildContext context, bool visitor(InheritedElement element)) {
+    InheritedElement actionsElement = context.getElementForInheritedWidgetOfExactType<_ActionsMarker>();
+    while (actionsElement != null) {
+      if (visitor(actionsElement) == true) {
+        break;
+      }
+      // _getParent is needed here because
+      // context.getElementForInheritedWidgetOfExactType will return itself if it
+      // happens to be of the correct type.
+      final BuildContext parent = _getParent(actionsElement);
+      actionsElement = parent.getElementForInheritedWidgetOfExactType<_ActionsMarker>();
+    }
+    return actionsElement != null;
+  }
 
   // Finds the nearest valid ActionDispatcher, or creates a new one if it
   // doesn't find one.
-  static ActionDispatcher _findDispatcher(Element element) {
-    assert(element.widget is Actions);
-    final Actions actions = element.widget as Actions;
-    ActionDispatcher dispatcher = actions.dispatcher;
-    if (dispatcher == null) {
-      bool visitAncestorElement(Element visitedElement) {
-        if (visitedElement.widget is! Actions) {
-          // Continue visiting.
-          return true;
-        }
-        final Actions actions = visitedElement.widget as Actions;
-        if (actions.dispatcher == null) {
-          // Continue visiting.
-          return true;
-        }
-        dispatcher = actions.dispatcher;
-        // Stop visiting.
-        return false;
+  static ActionDispatcher _findDispatcher(BuildContext context) {
+    ActionDispatcher dispatcher;
+    _visitActionsAncestors(context, (InheritedElement element) {
+      final ActionDispatcher found = (element.widget as _ActionsMarker).dispatcher;
+      if (found != null) {
+        dispatcher = found;
+        return true;
       }
+      return false;
+    });
+    return dispatcher ?? const ActionDispatcher();
+  }
 
-      element.visitAncestorElements(visitAncestorElement);
+  /// Returns a [VoidCallback] handler that invokes the bound action for the
+  /// given `intent` if the action is enabled, and returns null if the action is
+  /// not enabled.
+  ///
+  /// This is intended to be used in widgets which have something similar to an
+  /// `onTap` handler, which takes a `VoidCallback`, and can be set to the
+  /// result of calling this function.
+  ///
+  /// Creates a dependency on the [Actions] widget that the bound action is
+  /// found in so that if the actions change, the context will be rebuilt and
+  /// find the updated action. If you cache the result of this function, you
```

Comment:
nit: I would (maybe in a side sentence) emphasize that caching this is not really necessary and that it is ok to just call this every build? The method seems reasonably cheap? (Otherwise I am afraid that some users will read this as "I should really cache these)

Reply:
Maybe the whole sentence about refreshing the value can be removed? I mean, that's really just restating what `didChangeDependencies` is for.

I took it out.  Let me know if that is better, or if I need to explain it here.